test/test-lib/contracts-mock.ts

```diff
@@ -0,0 +1,42 @@
+import * as nock from 'nock';
+import * as path from 'path';
+
+export const mockRepo = (
+	owner: string,
+	name: string,
+	filename?: string,
+	persist = false,
+) => {
+	const redirectInterceptor = nock('https://api.github.com')
+		.persist(persist)
+		.get(`/repos/${owner}/${name}/tarball`);
+
+	redirectInterceptor.reply(302, undefined, {
+		Location: `https://codeload.github.com/${owner}/${name}/legacy.tar.gz/master`,
+	});
+
+	const fileInterceptor = nock('https://codeload.github.com/')
+		.persist(persist)
+		.get(`/${owner}/${name}/legacy.tar.gz/master`);
+
+	fileInterceptor.replyWithFile(
+		200,
+		path.join(__dirname, `../fixtures/contracts/${filename ?? name}.tar.gz`),
```

Comment:
How big are these contracts? In general I'd appreciate it if you could make them a text file that we compress on the fly here. That'll make viewing/editing/diff'ing them much easier